components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -5384,4 +5387,180 @@ protected String getSearchQuery(String searchQuery) throws APIManagementExceptio
+                    String[] faultSeqChildPaths = defaultSeqCollection.getChildren();
+                    for (String defaultSeqChildPath : faultSeqChildPaths) {
+                        Resource defaultSequence = registry.get(defaultSeqChildPath);
+                        try {
+                            OMElement seqElment = APIUtil.buildOMElement(defaultSequence.getContentStream());
+                            if (name.equals(seqElment.getAttributeValue(new QName("name")))) {
+                                return defaultSequence;
+                            }
+                        } catch (OMException e) {
+                            log.error("Error occurred when reading the sequence '" + defaultSeqChildPath +
+                                    "' from the registry.", e);
+                        }
+                    }
+                }
+            }
+        }  catch (RegistryException e) {
+            throw new APIManagementException("Error while retrieving registry for tenant " + tenantId, e);
+        } catch (org.wso2.carbon.registry.api.RegistryException e) {
+            throw new APIManagementException("Error while processing the " + defaultSequenceFileLocation +
+                    " in the registry", e);
+        } catch (Exception e) {
+            throw new APIManagementException(e.getMessage(), e);
+        }
+        return null;
+    }
+
+    /**
+     * Get the resource which matches the user selected resource type and the name from the custom uploaded sequences.
+     * @param identifier : The API Identifier.
+     * @param type : The sequence type.
+     * @return : Resource object which matches the parameters. If no resource found, return null.
+     */
+    private Resource getCustomSequence(APIIdentifier identifier, String type, String name) throws
+            APIManagementException {
+        Resource customSequence = null;
+        boolean isTenantFlowStarted = false;
+        try {
+            String tenantDomain = null;
+            if (identifier.getProviderName().contains("-AT-")) {
+                String provider = identifier.getProviderName().replace("-AT-", "@");
+                tenantDomain = MultitenantUtils.getTenantDomain(provider);
+            }
+            PrivilegedCarbonContext.startTenantFlow();
+            isTenantFlowStarted = true;
+            if (!StringUtils.isEmpty(tenantDomain)) {
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
+            } else {
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain
+                        (MultitenantConstants.SUPER_TENANT_DOMAIN_NAME, true);
+            }
+            UserRegistry registry = ServiceReferenceHolder.getInstance().getRegistryService()
+                    .getGovernanceSystemRegistry(tenantId);
+
+            String customSeqFileLocation = "";
+            if ("fault".equals(type)) {
```

Comment:
better to move these strings to constants

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/5114/commits/c6bc99c70b7ef3df0181e7565b77152ab729b066